java/gateway-core/src/main/java/com/avaulta/gateway/rules/ParameterSchemaUtils.java

```diff
@@ -0,0 +1,87 @@
+package com.avaulta.gateway.rules;
+
+import com.avaulta.gateway.pseudonyms.impl.UrlSafeTokenPseudonymEncoder;
+import org.apache.commons.lang3.tuple.Pair;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.regex.Pattern;
+
+/**
+ * utils for validating values conform to parameter schemas
+ *
+ * NOTE: unlike JsonSchemaFilterUtils, as that *redacts* values that don't match schema; this validates that a value
+ * matches schema, more analogous to how standard JsonSchema is meant to be used.
+ *
+ */
+public class ParameterSchemaUtils {
+
+
+    /**
+     * validate that a value conforms to a schema
+     * @param schema to validate against
+     * @param value to validate
+     * @return whether value conforms to schema
+     */
+    public boolean validate(ParameterSchema schema, String value) {
+        if (value != null) {
+            if (schema.getType() != null) {
+                if (schema.getType().equals(ParameterSchema.ValueType.INTEGER.getEncoding())) {
+                    try {
+                        Integer.parseInt(value);
```

Comment:
I'd be safe and do `Long.parseLong`, things can easily go beyond 2³² in autogenerated ids.